ruma-api-macros/src/api/request.rs

```diff
@@ -173,12 +173,25 @@ impl Request {
 
                 quote_spanned! {span=>
                     #( #cfg_attrs )*
-                    #field_name: #src.#field_name
+                    #field_name: #src.#field_name,
                 }
             })
-        });
+        };
+
+        let mut fields = vec![];
```

Comment:
It seems a bit weird to fill this `Vec` with `Option`s. If you use `extend` instead of `push`, only `Some`s are added.